src/jld.jl

```diff
@@ -166,7 +163,7 @@ function serialize_internal(x::Union{Dict,IdDict}, stackdict::IdDict)
         return (stackdict[x] = x)
     end
 
-    dest = Dict()
+    dest = x <: Dict ? Dict() : IdDict()
```

Comment:
This may not recover the original type, as there are more cases. Maybe you should store the original type as well?

```
julia> subtypes(AbstractDict)
8-element Array{Any,1}:
 Base.EnvDict
 Base.ImmutableDict
 Base.Iterators.Pairs
 Dict
 IdDict
 OrderedCollections.OrderedDict
 Test.GenericDict
 WeakKeyDict
```

Reply:
I intentionally did this. For example if the input is Dict{String,KnetArray} , we cannot save it as Dict{String,KnetJLD}

Reply:
I understand.  However when we save/load some data we should recover all the original types.  Thus my suggestion of "store the original types as well", maybe as part of a struct?